### ISSUE TITLE
non-numeric axes updates

### ISSUE THREAD

**Issue by faizan-khan-iit** [tdhock/animint#174]
Last week, I was working with some data, and I realized that we have no system in place to check whether the columns in the data to be used for axes updates are numeric or not. Refer to the example below:
```
library(animint)
d <- mtcars
d$cyl <- as.factor(d$cyl)
# Convert one of the data columns to factor
d$disp <- as.factor(d$disp)

# No updates
plot_a <- ggplot() + geom_point(aes(x=hp, y=disp, colour=cyl), data = d)
viz <- list(a = plot_a)
animint2dir(viz, "./test_plots")

# With updates. Note: Y axis has non-numeric values in data
plot_a_update_xy <- plot_a + theme_animint(update_axes = c("x", "y"))
viz_update <- list(a=plot_a_update_xy)
viz_update$selector.types = list(cyl="single")
animint2dir(viz_update, "./test_plots2")
``` 

As you can see, the second animint does not load properly. I think this is because we use the built data to calculate the axes domains, and they are modified by `ggplot_build`.

Built data when both axis values are numeric:
```
    x   y
1 110 160
2 110 160
3  93 108
4 110 258
5 175 360
6 105 225
```
Built data when Y axis is coerced to factors:
```
    x  y
1 110 13
2 110 13
3  93  6
4 110 16
5 175 23
6 105 15
```

So this messes up our logic a bit.
I think that it doesn't make sense to have axes updates for non-numeric data. We should probably just stop with an error and inform the user to make sure the values are numeric, in case this happens.

**Comment by faizan-khan-iit** [comment]
If this is to be done, it should be easy enough to implement. I will go ahead and add a test for this.

**Comment by faizan-khan-iit** [comment]
Closed by #175
